controllers/storagesystem_controller.go

```diff
@@ -80,6 +81,8 @@ func (r *StorageSystemReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 
 	logger.Info("storagesystem instance found")
 
+	metrics.ReportODFSystemMapMetrics(instance.Name, instance.Spec.Name, instance.Spec.NameSpace, string(instance.Spec.Kind))
```

Comment:
Do we need this every reconcile? Can not we move this to if condition where we are adding a finalizer. So we can just call it for the first time.

Reply:
If `Spec` is immutable, then we could do that. Else, we need this on every reconcile to account for changes.